packages/sit-onyx/src/composables/scrollEnd.ts

```diff
@@ -0,0 +1,67 @@
+import { ref, unref, watchEffect, type Directive, type MaybeRef, type Ref } from "vue";
+
+export type UseScrollEndOptions = {
+  /**
+   * Whether currently loading. Will pause scroll listeners if `true`.
+   */
+  loading: Ref<boolean | undefined>;
+  /**
+   * Whether scroll events should be captured.
+   * Can be disabled if not needed to improve performance.
+   *
+   * @default true
+   */
+  enabled?: Ref<boolean>;
+  /**
+   * Offset (in pixel). Can be used to trigger the callback earlier (e.g. if scrolled to second last option).
+   */
+  offset?: MaybeRef<number | undefined>;
+};
+
+/**
+ * Directive used for lazy loading which will keep track of whether the component is scrolled to the end (vertically).
+ */
+export const useScrollEnd = (options: UseScrollEndOptions) => {
+  /**
+   * We have to check if the scroll amount is close enough to some threshold in order to
+   * more accurately calculate arrivedState. This is because scrollTop/scrollLeft are non-rounded
+   * numbers, while scrollHeight/scrollWidth and clientHeight/clientWidth are rounded.
+   * https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight#determine_if_an_element_has_been_totally_scrolled
+   */
+  const ARRIVED_STATE_THRESHOLD_PIXELS = 1 as const;
+
+  /**
+   * Whether the component is scrolled to the end (vertically).
+   */
+  const isScrollEnd = ref(false);
+
+  /**
+   * Updates the isScrollEnd depending on the event targets scroll position.
+   */
+  const handleScroll = (event: Event) => {
+    const target = event.target;
+    if (!(target instanceof Element)) return;
+
+    const offset = unref(options.offset) ?? 0;
```

Comment:
Should we be so nice to really make sure devs can not use it wrong and handle negative offset input too? ^^

```suggestion
    const offset = Math.abs(unref(options.offset) ?? 0);
```

because typescript can not inform the dev that we expect > 0 numbers and currently a negative number will make it unusable

Reply:
are fraction offset numbers OK too?